main/lsp/LSPFileUpdates.h

```diff
@@ -25,6 +25,18 @@ class LSPFileUpdates final {
 
     TypecheckingPath typecheckingPath = TypecheckingPath::Slow;
 
+    // The names of all the changed files involved in a fast path edit, and the index of their data in the
+    // `updatedFiles` vector. This vector stores paths because file refs can change between deciding if it's possible to
+    // take the fast path and actually taking it.
+    std::vector<std::pair<std::string, size_t>> fastPathChangedFiles;
+
+    // Indicates whether or not the incremental namer should be used on the fast path.
+    bool fastPathUseIncrementalNamer = false;
+
+    // The paths of any additional files implicated in a fast path edit. This vector stores paths because file refs can
+    // change between deciding if it's possible to take the fast path and actually taking it.
+    std::vector<std::string> fastPathExtraFiles;
```

Comment:
```suggestion
    // The paths of any additional files implicated in a fast path edit. This vector stores path
    // strings because FileRef IDs might be different in the GlobalState used on the indexing thread
    // and the typechecking thread, but the `LSPFileUpdates` structure must be valid to transfer
    // ownership from one thread to the other.
    std::vector<std::string> fastPathExtraFiles;
```

Reply:
Note that one thing which would disprove my comment here is that the LSPFileUpdate already stores `updatedFileIndexes`, which has parsed trees, which have NameRefs. I think that suggests that those trees can only be used on the indexer thread, not the typechecking thread? (or maybe vice versa?)

It would be good to figure that out, and then leave a comment on `updatedFileIndexes`.

Reply:
`updatedFileIndexes` is definitely only meant to be used with `initialGS`, and it's moved through onto the typechecker's `indexed` vector, which has another note to this effect already on it. The fast path uses the typechecker's global state, so I'm pretty sure that your comment is valid. I've added a note to `updatedFileIndexes` to indicate that it's tied to `initialGS`.

Reply:
ah cool, I see it now. that's the comment I was looking to try to show you the other day

https://github.com/sorbet/sorbet/blob/8fea7d8b1d714ac240a544c25bc495d430c793bc/main/lsp/LSPTypechecker.cc#L297-L299

Reply:
Ah, that's a helpful comment!